block/sync.go

```diff
@@ -92,16 +87,18 @@ func (m *Manager) SettlementSyncLoop(ctx context.Context) error {
 
 				err = m.attemptApplyCachedBlocks()
 				if err != nil {
-					uevent.MustPublish(context.TODO(), m.Pubsub, &events.DataHealthStatus{Error: err}, events.HealthStatusList)
+					m.freezeNode(context.Background(), err)
```

Comment:
same here. imo freeze should be called from callsite if possible.

Reply:
updated it